### ISSUE TITLE
Add option to prevent transcoding of audio files

### ISSUE THREAD

**Issue by inorick** [ytdl-org/youtube-dl#28006]
## Checklist

- [x] I'm reporting a feature request
- [x] I've verified that I'm running youtube-dl version **2021.01.24.1**
- [x] I've searched the bugtracker for similar feature requests including closed ones

## Description

Downloading a lossless audio file (tested with .wav and .aiff) with the `-x` flag causes the audio to be transcoded into mp3. For archiving purposes, having the option to prevent all (audio) transcoding would be ideal.

Examples links:
https://soundcloud.com/caleesi-sarahkreis/caleesi-sarahkreis-mbr-sao-paulo
https://soundcloud.com/caleesi/arutani-caleesi-ennui-original-mix

**Comment by s104346** [comment]
Don't use `-x` then, just `youtube-dl https://soundcloud.com/...`
The description for `-x` even says: "**Convert** video files to audio-only files".

**Comment by inorick** [comment]
Not using `-x` indeed gives me the uncompressed. Thanks for that.
I still do not see why `-x` causes a conversion given the fact the there was no video present in the first place. Shouldn't it fall back to leaving audio-only files as is?

**Comment by rautamiekka** [comment]
What's your whole command line ?

**Comment by inorick** [comment]
Here is the command:
```
$ youtube-dl -x https://soundcloud.com/caleesi-sarahkreis/caleesi-sarahkreis-mbr-sao-paulo
[soundcloud] caleesi-sarahkreis/caleesi-sarahkreis-mbr-sao-paulo: Downloading info JSON
[soundcloud] 778172407: Downloading JSON metadata
[soundcloud] 778172407: Downloading webpage
[soundcloud] 778172407: Downloading JSON metadata
[soundcloud] 778172407: Downloading JSON metadata
[soundcloud] 778172407: Downloading JSON metadata
[download] Destination: Caleesi & Sarah Kreis _ MBR São Paulo _ February 2020-778172407.aiff
[download] 100% of 1.73GiB in 02:02
[ffmpeg] Destination: Caleesi & Sarah Kreis _ MBR São Paulo _ February 2020-778172407.mp3
Deleting original file Caleesi & Sarah Kreis _ MBR São Paulo _ February 2020-778172407.aiff (pass -k to keep)
```

The `[ffmepg]` step takes some time to complete during which `htop` shows the transcoding process:
![cpu](https://user-images.githubusercontent.com/27500920/106336462-3e4cd480-628f-11eb-8e27-cd4b8d4fc3d1.png)

**Comment by pukkandan** [comment]
`-x` not only extracts the audio, but converts it into a known audio-only format (mp3, ogg etc)

**Comment by inorick** [comment]
It even converts `.wav` files (see below). Is there a list of formats that `-x` converts and what formats it leaves untouched?

```
/youtube-dl.exe -x https://soundcloud.com/caleesi/arutani-caleesi-ennui-original-mix
[soundcloud] caleesi/arutani-caleesi-ennui-original-mix: Downloading info JSON
[soundcloud] 427069410: Downloading JSON metadata
[soundcloud] 427069410: Downloading webpage
[soundcloud] 427069410: Downloading JSON metadata
[soundcloud] 427069410: Downloading JSON metadata
[soundcloud] 427069410: Downloading JSON metadata
[download] Destination: Arutani & Caleesi - Ennui (Original Mix)-427069410.wav
[download] 100% of 119.84MiB in 00:08
[ffmpeg] Destination: Arutani & Caleesi - Ennui (Original Mix)-427069410.mp3
Deleting original file Arutani & Caleesi - Ennui (Original Mix)-427069410.wav (pass -k to keep)
```

**Comment by Vangelis66** [comment]
> Is there a list of formats that -x converts and **what formats it leaves untouched**?

As @pukkandan wrote, `-x` flag **always** transcodes/converts audio; additionally, from `Help`: 

``` 
   --audio-format FORMAT                 Specify audio format: "best", "aac",
                                         "flac", "mp3", "m4a", "opus", "vorbis",
                                         or "wav"; "best" by default; No effect
                                         without -x
```
Perhaps an additional value `remux` (or something to that end...) could be added to `--audio-format`, to achieve what you want... Or just NOT use `-x` (as already instructed) or, if you do, use `-x -k` to preserve the original audio codec... 
In conclusion, **soundcloud.com** is obviously only serving AUDIO encodes, thus **I see no reason whatsoever for using the '-x` flag in the firat place**, which is solely intended for media files comprised of both Video+Audio elementary streams...

**Comment by inorick** [comment]
> As @pukkandan wrote, -x flag always transcodes/converts audio;
This is not completely true as using `-x` on an `.mp3` stream will not cause transcoding:
```
youtube-dl.exe -x https://soundcloud.com/talderverwirrung/caleesi-im-tal-057
[soundcloud] talderverwirrung/caleesi-im-tal-057: Downloading info JSON
[...]
[download] Destination: Caleesi Im Tal - #057-956484631.mp3
[download] 100% of 56.93MiB in 00:04
[ffmpeg] Post-process file Caleesi Im Tal - #057-956484631.mp3 exists, skipping
```
It looks like the rule of the flag is "strip video and transcode audio unless it is mp3".

The current behavior makes it harder to use `youtube-dl` in a script to extract audio (with minimal transcoding) that targets multiple sources (e.g. soundcloud + youtube). It would require to set the `-x` flag for video sites but omit it for audio-only sites.

Having a `demux` option looks like a way to address this. I would even argue that it should be the default since the name of the `--extract-audio` flag suggests extraction only. Transcoding should be an optional second step as it decreases quality further.

**Comment by Vangelis66** [comment]
> It looks like the current rule of the flag is "**strip video and transcode audio unless it is mp3**"

Yes :+1: , the logic here being "best" (default) = "mp3", as you've found yourself already (probably because, despite its old age and several shortcomings, MP3 is still the most popular and widely hardware-compatible audio codec, especially after its patent has expired...) .

> (e.g. soundcloud + youtube)

For `youtube`, if **audio** is ALL you focus on, you can explicitly ask for audio-only yt streams and do away with `-x` altogether...

> Having a demux option looks like a way to address this. I would even argue that it should be the default since the name of the --extract-audio flag suggests extraction only. Transcoding should be an optional second step as it decreases quality further.

I agree with you the current wording of the `--extract-audio=-x` option is not transparent enough as to what actually is being done and there's certainly room for improvement there; `Help` documents that:  
```
     -x, --extract-audio                 Convert video files to audio-only files
                                         (requires ffmpeg/avconv and
                                         ffprobe/avprobe)
```
so that needs to be expanded on [e.g. "Demux and transcode (by default to MP3) **Audio** stream from V+A media file"] and the verbose switch maybe changed to `--extract-transcode-audio` ...

But your request is a legitimate one and one I second myself... Perhaps a `--demux-audio` new switch could be created for that purpose, but then one should consider the fact elementary audio streams may have to be muxed again losslessly into a proper container; your **soundcloud** example, I'll be frank, **isn't a valid consideration here**, `--extract-audio`, `--extract-transcode-audio`, `--extract-convert-audio`, `--demux-audio` or whatever would only target media files with both V+A content...  :wink: 

In any case, what the devs would most like is a (proper) **Pull Request**, so if anyone with the know-how is inclined, chime in... :smile: 

Best regards :smile_cat:

**Comment by pukkandan** [comment]
I looked through the source code of extractaudio. These are the conversions it does when you don't give --audio-format:

aac > m4a
flac, mp3, opus = untouched
vorbis > ogg
everything else > mp3
